<launch>
    <!-- Default values - you don't have to change this if using start.py. -->
    <arg name="num_of_robots" default="6"/>         <!-- Number of robots used. -->
    <arg name="map_name" default="test_10x10"/>     <!-- Name of the map used. -->
    <arg name="start_rviz" default="true"/>        <!-- Automatically start Rviz. ***************** false--> 

    <arg name="map_world" default="$(find sphero_stage_2)/resources/worlds/$(arg map_name)_$(arg num_of_robots).world"/>
    <arg name="map_yaml" default="$(find sphero_stage_2)/resources/maps/$(arg map_name)/$(arg map_name).yaml"/>

    <!-- Set arguments as ros parameter so all nodes can access them. -->
    <param name="num_of_robots" type="int" value="$(arg num_of_robots)"/>
    <param name="robot_name" type="string" value="robot"/>
    <param name="run_type" type="string" value="sim"/>
    <param name="data_associated" type="bool" value="true"/>

    <!-- Start map server. -->
    <node name="world2ned" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 map odom 100"/>  

    <!-- <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/> -->
    <node pkg="map_server" type="map_server" args="$(arg map_yaml)" name="map_server"/>

    <!-- Start Stage simulator. -->
    <node pkg="stage_ros" type="stageros" name="simulator" args="$(arg map_world)"/>

    <!-- Start simulation_tf node: provide tf transforms for simulation. -->
    <!-- <node pkg="sphero_stage_2" type="simulation_tf.py" name="tf_server"/> *****************  -->
    <node pkg="sphero_stage_2" type="simulation_tf_modified.py" name="tf_server"/>   
    <!-- <node pkg="sphero_stage_2" type="obstacle_avoidance_map_working.py" name="cluster_generation"/>    -->
    <node pkg="sphero_stage_2" type="robot_boundary.py" name="robot_search_area_"/>   
    <!-- <node pkg="sphero_stage_2" type="visual_rviz.py" name="visualize_centroids"/>    -->
    <!-- <node pkg="sphero_stage_2" type="visual_robot_vector.py" name="visualize_velocity_vector"/>     -->
    <node pkg="sphero_stage_2" type="visual_robot_vector_many.py" name="visualize_velocity_vector"/>    

    <!-- Start rviz. -->
    <!-- <group if="$(arg start_rviz)"> -->
    <param name="robot_description" textfile="$(find 2_sphero_description)/urdf/simple_ball.urdf"/>


    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find sphero_stage_2)/launch/sphero_sim.rviz"/>
    <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find sphero_stage_2)/launch/sphero_sim_1.rviz"/> -->
    <!-- <node pkg="rviz" type="rviz" name="rviz"/> -->
    <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find sphero_stage_2)/launch/sphero_sim_$(arg num_of_robots)_robots.rviz"/> -->
    <!-- </group> -->
</launch>
